src/app/i18n/services/i18n.service.ts

```diff
@@ -1,20 +1,40 @@
-import _ from 'lodash';
-import format from 'string-template';
-import { Locale } from '../types';
+import { initReactI18next } from 'react-i18next';
+import i18next, { TFunctionDetailedResult } from 'i18next';
+import { useTranslation } from 'react-i18next';
+import localStorageService from 'app/core/services/local-storage.service';
 
-import locales from '../locales';
+let deviceLang = navigator.language.split('-')[0];
 
-class I18nService {
-  currentLocale: string = Locale.English;
+if (localStorageService.get('language')) {
+  deviceLang = localStorageService.get('language') as string;
+} else {
+  localStorageService.set('language', deviceLang);
+}
 
-  get(key: string, values = {}): string {
-    const messageTemplate = _.get(locales[this.currentLocale], key) || key;
-    const result = format(messageTemplate, values);
+i18next
+  .use(initReactI18next) // passes i18n down to react-i18next
+  .init({
+    resources: {
+      en: {
+        translation: require('../locales/en.json'),
+      },
+      es: {
+        translation: require('../locales/es.json'),
+      },
+    },
+    fallbackLng: 'en',
+    lng: deviceLang,
+    defaultNS: 'translation',
+    ns: ['translation'],
+    interpolation: {
+      escapeValue: false, // not needed for react as it escapes by default
+    },
+  });
 
-    return result;
-  }
+export function get(value: string, object?: Record<string, unknown>): string {
+  const { t } = useTranslation();
+  const result = t(value, object as TFunctionDetailedResult) as string;
+  return result;
 }
 
```

Comment:
Insted of use the useTranslation of react-i18next library directly around all the app, would be nice to make an abstraction to improve maintenance of the code. 

Could add a `TrasnlationsProvider.tsx` that uses react-18next. Here you have a sugestion:
```suggestion
import React, { createContext, useContext, useMemo } from 'react';
import { useTranslation } from 'react-i18next';

interface TranslationContextProps {
  translate: (key: string) => string;
}

const TranslationContext = createContext<TranslationContextProps>({ translate: () => '' });

interface TranslationProviderProps {
  children: React.ReactNode;
}

export const TranslationProvider: React.FC<TranslationProviderProps> = ({ children }) => {
  const { t } = useTranslation();
  const value = useMemo(() => ({ translate: t }), [t]);

  return <TranslationContext.Provider value={value}>{children}</TranslationContext.Provider>;
};

export const useTranslationContext = (): TranslationContextProps => {
  return useContext(TranslationContext);
};

```
With this you could use the provider to wrap the app and that let you use the useTranslationContext as hook
```
ReactDOM.render(
  <React.StrictMode>
    <Provider store={store}>
      <TranslationProvider>
        <App />
      </TranslationProvider>
    </Provider>
  </React.StrictMode>,
  document.getElementById('root'),
);
```

If you gonna use this code feel free to make any modification and change names of that you consider